testsuites/listener/shared/client_sg/listener_initial_replication.py

```diff
@@ -0,0 +1,93 @@
+from keywords.utils import log_info
+from keywords.utils import log_r
+from keywords.MobileRestClient import MobileRestClient
+from libraries.data import doc_generators
+
+
+def large_initial_pull_replication(ls_url, cluster_config, num_docs, continuous):
+
+    sg_db = "db"
+    ls_db = "ls_db"
+
+    sg_one_admin = cluster_config["sync_gateways"][0]["admin"]
+    sg_one_public = cluster_config["sync_gateways"][0]["public"]
+
+    log_info(ls_url)
+    log_info(sg_one_admin)
+    log_info(sg_one_public)
+
+    client = MobileRestClient()
+    client.create_user(sg_one_admin, sg_db, "seth", password="password", channels=["ABC", "NBC"])
+    session = client.create_session(sg_one_admin, sg_db, "seth")
+
+    # Create 10000 docs on sync_gateway
+    docs = client.add_docs(
+        url=sg_one_public,
+        db=sg_db,
+        number=num_docs,
+        id_prefix="seeded_doc",
+        generator="four_k",
+        auth=session
+    )
+
+    client.create_database(url=ls_url, name=ls_db)
+
+    # Start oneshot pull replication
+    repl_id = client.start_replication(
+        url=ls_url,
+        continuous=continuous,
+        from_url=sg_one_admin,
+        from_db=sg_db,
+        to_db=ls_db
+    )
+
```

Comment:
I have polling loops on my verification functions (i.e. the will keep looking for expected docs until the find them all or timeout). Additionally, there is a `wait_for_replication_idle(replication_id)` method. Do you prefer one over the other?